packages/browser-repl/README.md

```diff
@@ -31,14 +31,20 @@ const runtime = new IframeRuntime(serviceProvider);
 Shell is a React component with the following properties:
 
 - `runtime: Runtime`: The runtime used to evaluate code.
-- `onOutputChanged?: (output: readonly ShellOutputEntry[]) => void`: A function called each time the output changes with an array of `ShellOutputEntryes`.
-- `onHistoryChanged?: (history: readonly string[]) => void`: A function called each time the history changes with an array of history entries ordered from the most recent to the oldest entry.
+- `onOutputChanged?: (output: ShellOutputEntry[]) => void`: A function called each time the output changes with an array of `ShellOutputEntries`.
+- `onHistoryChanged?: (history: string[]) => void`: A function called each time the history changes with an array of history entries ordered from the most recent to the oldest entry.
+- `onEditorChanged?: (editor: EditorRef | null) => void`: A function called each time the editor ref changes. Can be used to call editor methods.
+- `onOperationStarted?: () => void`: A function called when an operation has begun.
+- `onOperationEnd?: () => void`: A function called when an operation has completed (both error and success).
 - `redactInfo?: boolean`: If set, the shell will omit or redact entries containing sensitive info from history. Defaults to `false`.
 - `maxOutputLength?: number`: The maxiumum number of lines to keep in the output. Defaults to `1000`.
 - `maxHistoryLength?: number`: The maxiumum number of lines to keep in the history. Defaults to `1000`.
-- `initialOutput?: readonly ShellOutputEntry[]`: An array of entries to be displayed in the output area. Can be used to restore the output between sessions, or to setup a greeting message. **Note**: new entries will not be appended to the array.
-- `initialHistory?: readonly string[]`: An array of history entries to prepopulate the history.
+- `initialEvaluate?: string|string[]`: A set of input strings to evaluate right after shell is mounted.
+- `inputText?: string`: Initial text for the input field.
```

Comment:
How do you feel about prefixing this with initial, `initialInputText`?

Reply:
The way I see it it isn't an initial value, it is the controlled value. Same as a leafygreen text input's `value` or a modal's `open`. But I also don't feel too strongly about it.

I should update the readme line, though.

Reply:
OK I'm going with initialText since that's what we pass down to the input.